bundles/org.openhab.binding.mqtt.homeassistant/src/test/java/org/openhab/binding/mqtt/homeassistant/internal/HAConfigurationTests.java

```diff
@@ -69,14 +69,21 @@ public void testAbbreviations() {
         assertThat(config.payload_not_available, is("G"));
 
         assertThat(config.device, is(notNullValue()));
-        assertThat(config.device.identifiers, contains("H"));
-        assertThat(config.device.connections, is(notNullValue()));
-        assertThat(config.device.connections.get(0).type, is("I1"));
-        assertThat(config.device.connections.get(0).identifier, is("I2"));
-        assertThat(config.device.name, is("J"));
-        assertThat(config.device.model, is("K"));
-        assertThat(config.device.sw_version, is("L"));
-        assertThat(config.device.manufacturer, is("M"));
+
+        BaseChannelConfiguration.Device device = config.device;
+        if (device != null) {
+            assertThat(device.identifiers, contains("H"));
+            assertThat(device.connections, is(notNullValue()));
+            List<@NonNull Connection> connections = device.connections;
+            if (connections != null) {
+                assertThat(connections.get(0).type, is("I1"));
+                assertThat(connections.get(0).identifier, is("I2"));
+            }
+            assertThat(device.name, is("J"));
+            assertThat(device.model, is("K"));
+            assertThat(device.sw_version, is("L"));
+            assertThat(device.manufacturer, is("M"));
+        }
```

Comment:
```
} else { 
    Assert.fail("device is null");
}
```

The test failed for a null value before

Reply:
Yes, the tests already failed in line 71:
```
 assertThat(config.device, is(notNullValue()));
```
so the else branch is never reached.

Reply:
Strange. That line showed as removed when I looked at it.